### PR TITLE
Limit non-node peer connections to 100

### PR DESCRIPTION
**Pull request by alexpan0610**

### CODE REVIEW

File: elanet/server.go [hunk 1]
```diff
@@ -1003,3 +1024,8 @@ func makeEmptyMessage(cmd string) (p2p.Message, error) {
 	}
 	return message, nil
 }
+
+// nodeFlag returns if a peer contains the full node flag.
+func nodeFlag(flag uint64) bool {
+	return pact.ServiceFlag(flag)&pact.SFNodeNetwork != pact.SFNodeNetwork
```
Comment (raindust) [inline elanet/server.go]: should be "pact.ServiceFlag(flag)&pact.SFNodeNetwork == pact.SFNodeNetwork"?

File: elanet/server.go [hunk 2]
```diff
@@ -116,18 +121,34 @@ func (sp *serverPeer) handleDisconnect() {
 	sp.WaitForDisconnect()
 	sp.server.routes.DonePeer(sp.Peer)
 	sp.server.syncManager.DonePeer(sp.Peer)
+
+	// Decrease non node count.
+	if !nodeFlag(uint64(sp.Services())) {
```
Comment (raindust) [inline elanet/server.go]: reconsider condition

File: elanet/server.go [hunk 3]
```diff
@@ -116,18 +121,34 @@ func (sp *serverPeer) handleDisconnect() {
...
+	}
 }
 
 // OnVersion is invoked when a peer receives a version message and is
 // used to negotiate the protocol version details as well as kick start
 // the communications.
 func (sp *serverPeer) OnVersion(_ *peer.Peer, m *msg.Version) {
 	// Disconnect full node peers that do not support DPOS protocol.
-	if m.Relay && sp.ProtocolVersion() < pact.DPOSStartVersion {
+	if nodeFlag(m.Services) && sp.ProtocolVersion() < pact.DPOSStartVersion {
```
Comment (raindust) [inline elanet/server.go]: reconsider condition

File: elanet/server.go [hunk 4]
```diff
@@ -116,18 +121,34 @@ func (sp *serverPeer) handleDisconnect() {
...
 		sp.Disconnect()
 		return
 	}
 
+	if !nodeFlag(m.Services) {
```
Comment (raindust) [inline elanet/server.go]: reconsider condition

File: elanet/server.go (hunk 1 above)
Comment (alexpan0610) [reply elanet/server.go]: Yeah, my mistake.

**Pull request merged**